### ISSUE TITLE
Panic in git repo with non-existent symlink

### ISSUE THREAD

**Issue by ethack** [ogham/exa#674]
I ran into a situation where I had a git repo which contained a symlink with a non-existent target. This caused exa to crash when run with `exa --long --git`. See below for details and reproduction steps.

```
$ exa --version   
exa v0.9.0

$ uname -a                        
Linux poprocks 5.4.0-7626-generic #30~1588169883~20.04~bbe668a-Ubuntu SMP Wed Apr 29 21:00:02 UTC  x86_64 x86_64 x86_64 GNU/Linux
```

This sets up the conditions I believe cause the issue.

```
$ mkdir test  

$ cd test

$ ln -s doesnotexist symlink            

$ exa --long  --git                                         
lrwxrwxrwx 12 user 20 May 20:43 symlink -> doesnotexist

$ git init  
Initialized empty Git repository in /home/user/test/.git/

$ exa --long                                    
lrwxrwxrwx 12 user 20 May 20:43 symlink -> doesnotexist
```

Here's where the error occurs:

```
$ exa --long --git
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/libcore/result.rs:999:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
[1]    489979 abort (core dumped)  exa --long --git

$ RUST_BACKTRACE=1 exa --long --git
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/libcore/result.rs:999:5
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: start_thread
  16: __clone
[1]    490792 abort (core dumped)  RUST_BACKTRACE=1 exa --long --git
```

This just shows that it works correctly with a symlink with a target that does exist.

```
$ rm symlink      

$ ln -s . symlink    

$ exa --long --git
lrwxrwxrwx 8 user 20 May 20:57 -- symlink -> .
```

**Comment by Freaky** [comment]
Duplicate of #642, #637 #629, #623, #605, #558, and #526.  It was fixed in #584, so just waiting on a new release.

**Comment by ethack** [comment]
My bad. I definitely should have looked for related issues :facepalm: 

FWIW, I built exa from master and confirmed the issue goes away in my environment.

I'm not a rust dev so I made a simple docker container to build exa. It's very basic but in case it's useful for anyone I've copied it below.

```dockerfile
FROM rust

WORKDIR /app

COPY Cargo.toml Cargo.lock ./
RUN cargo fetch

COPY . ./
RUN make

CMD ["bash"]
```

The binary ends up in `/app/target/release/exa` and you can copy it out with:

```
docker build -t exa .
docker create --name exa exa
docker cp exa:/app/target/release/exa ./exa
docker rm exa
docker rmi exa
```
